packages/block-editor/src/components/list-view/branch.js

```diff
@@ -88,20 +100,35 @@ function ListViewBranch( props ) {
 		isExpanded,
 	} = props;
 
+	const isContentLocked = useSelect(
+		( select ) => {
+			return !! (
+				parentId &&
+				select( blockEditorStore ).getBlockAttributes( parentId )?.lock
+					?.content
+			);
+		},
+		[ parentId ]
+	);
+	const processedBlocks = useMemo( () => {
```

Comment:
For the ListView component: what do you think of filtering the blocks at this point, so the returned filtered blocks are the ones we want to render. At the moment the logic is spread into ListBranch and ListBlock.

Reply:
Minor: maintaining `filteredBlocks` as a name sounds like something that communicates well its role here.

Reply:
I've pushed these changes. ListViewBlock does not require any knowledge of a "content lock" now :)